packages/xo-web/src/common/xo/index.js

```diff
@@ -2105,6 +2105,38 @@ export const editSr = (sr, { nameDescription, nameLabel }) =>
 export const rescanSr = sr => _call('sr.scan', { id: resolveId(sr) })
 export const rescanSrs = srs => Promise.all(map(resolveIds(srs), id => _call('sr.scan', { id })))
 
+export const toggleSrMaintenanceMode = sr => {
+  const id = resolveId(sr)
+  const method = sr.in_maintenance_mode ? 'disableMaintenanceMode' : 'enableMaintenanceMode'
+
+  return _call(`sr.${method}`, { id }).catch(async err => {
+    if (
+      incorrectState.is(err, {
+        property: 'vmsToShutdown',
+      })
+    ) {
+      const vmIds = err.data.expected
+      const vmNames = vmIds.map(vmId => store.getState().objects.all[vmId].name_label)
+      await confirm({
+        title: _('maintenanceMode'),
+        body: (
+          <div>
+            {_('maintenanceSrModalBody', { n: vmNames.length })}
+            <ul>
+              {vmNames.map(name => (
+                <li key={name}>{name}</li>
+              ))}
```

Comment:
If there are 100 VMs to display, what should we do in this case? is it necessary to display all the VM names?

Reply:
I think it is important to display the affected VMs.

Reply:
Let's set a limit to 30 and add a "and N more" if there are more. IMO, if there are more than 30 VMs that will be shut down, either it doesn't matter to the user which VMs will be shut down (home lab, tests, ...) or they should thoroughly check from another view why there are so many VMs and if it's ok to shut them down. The list in the modal only serves as a final warning.

Reply:
We'll improve this in XO6 if necessary.